pymtl3/passes/backends/generic/structural/StructuralTranslatorL2.py

```diff
@@ -87,17 +87,17 @@ def recurse_struct_dtype_translation( dtype ):
   #-----------------------------------------------------------------------
 
   # Override
-  def rtlir_signal_expr_translation( s, expr, m ):
+  def rtlir_signal_expr_translation( s, expr, m, status = 'intermediate' ):
```

Comment:
how many possible status does it have? Maybe we just need `intermediate=True`?

Reply:
status is one of `reader`, `writer`, and `intermediate`. This argument is used to push the indices to the end of the expression

Reply:
i guess comparing long strings will make things slower ... up to you